cm/list.go

```diff
@@ -58,3 +69,41 @@ func (l list[T]) Data() *T {
 func (l list[T]) Len() uintptr {
 	return l.len
 }
+
+// MarshalJSON implements json.Marshaler.
+func (l list[T]) MarshalJSON() ([]byte, error) {
+	s := l.Slice()
+
+	if s == nil {
+		return nullLiteral, nil
+	}
+
+	// NOTE(lxf): Go JSON Encoder will serialize []byte as base64.
+	// We override that behavior so all int types have the same serialization format.
+	// []uint8{1,2,3} -> [1,2,3]
+	// []uint32{1,2,3} -> [1,2,3]
+	if byteArray, ok := any(s).([]byte); ok {
```

Comment:
This will miss `List[T]` where T is a named `uint8` type.

Try something like this and convert the underlying slice: https://go.dev/play/p/-MI8hTVGpGF

Reply:
makes sense! was avoiding the `unsafe.Pointer` route but totally forgot about type aliases.